### PR TITLE
[PW-1729] Updated Payment/Payout/Checkout APIs to v51

### PR DESCRIPTION
**Pull request by martinsrenato**
PAYMENTS
Added /getAuthenticationResult API and models needed for it
Added additional fields to existing models

MODIFICATION
Added /voidPendingRefund API and models needed for it
Added additional fields to existing models

PAYOUT
Separated version from payment/modification
Added additional fields to existing models

CHECKOUT
Added /paymentLinks API and models needed for it
Added additional fields to existing models

### CODE REVIEW

File: src/main/java/com/adyen/model/checkout/CheckoutPaymentsAction.java [hunk 1]
```diff
@@ -31,31 +31,91 @@
 import java.util.Map;
 import java.util.Objects;
 
+/**
+ * CheckoutPaymentsAction
+ */
+
+@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.JavaClientCodegen", date = "2019-11-20T13:58:41.085Z[GMT]")
```
Comment (rikterbeek) [inline src/main/java/com/adyen/model/checkout/CheckoutPaymentsAction.java]: please remove this.

Comment (martinsrenato) [reply src/main/java/com/adyen/model/checkout/CheckoutPaymentsAction.java]: Done

File: src/main/java/com/adyen/model/checkout/PaymentSessionRequest.java [hunk 2]
```diff
@@ -1383,195 +1462,102 @@ public boolean equals(java.lang.Object o) {
...
-                && Objects.equals(this.socialSecurityNumber, paymentSessionRequest.socialSecurityNumber)
-                && Objects.equals(this.splits, paymentSessionRequest.splits)
-                && Objects.equals(this.store, paymentSessionRequest.store)
-                && Objects.equals(this.storePaymentMethod, paymentSessionRequest.storePaymentMethod)
-                && Objects.equals(this.telephoneNumber, paymentSessionRequest.telephoneNumber)
-                && Objects.equals(this.threeDS2RequestData, paymentSessionRequest.threeDS2RequestData)
-                && Objects.equals(this.token, paymentSessionRequest.token)
-                && Objects.equals(this.totalsGroup, paymentSessionRequest.totalsGroup)
-                && Objects.equals(this.trustedShopper, paymentSessionRequest.trustedShopper)
-                && Objects.equals(this.uniqueTerminalId, paymentSessionRequest.uniqueTerminalId);
```
Comment (Aleffio) [inline src/main/java/com/adyen/model/checkout/PaymentSessionRequest.java]: uniqueTerminalId is still present in the file, should we add it back to the equals/hashcode methods or remove it from the file?

same for totalsGroup, accountInfo, additionalAmount, bankAccount, browserInfo, card, configId, deviceFingerprint, html, merchantRiskIndicator, mpiData, nationality, recurring, recurringProcessingModel, sdkVersion, selectedBrand, selectedRecurringDetailReference, sessionId, store, threeDS2RequestData, totalsGroup

Comment (martinsrenato) [reply src/main/java/com/adyen/model/checkout/PaymentSessionRequest.java]: I added it back, it was a mistake on my part when generating it.

File: src/main/java/com/adyen/service/resource/payment/GetAuthenticationResult.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+import com.adyen.Client;
+import com.adyen.Service;
+import com.adyen.service.Resource;
+
+public class GetAuthenticationResult extends Resource {
+
+    public GetAuthenticationResult(Service service) {
+        super(service,
+                service.getClient().getConfig().getEndpoint() + "/pal/servlet/Payment/" + Client.API_VERSION + "/getAuthenticationResult", null);
```
[the remark this answers is not in the document]
Comment (martinsrenato) [reply src/main/java/com/adyen/service/resource/payment/GetAuthenticationResult.java]: Done

**Pull request merged**